### PR TITLE
Add index page for "Tour of Code"

### PR DESCRIPTION
**Pull request by trickeydan**
See #18 

I have added all currently unarchived repositories in the SourceBots organisation on Github.

I am also aware that "Our Codebase" is perhaps not the most appropriate title for this section, open to suggestions.

The "More Info" column is intended to have links to pages with more information on the repo in the future, but as of yet, these pages have not been created.

### CODE REVIEW

File: content/codebase/_index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+| pkg         | {{< github "sourcebots/pkg" >}}         |           |
+| power-v4-hw | {{< github "sourcebots/power-v4-hw" >}} |           |
+| sym         | {{< github "sourcebots/sym" >}}         |           |
+
+## Misc
+
+| Name              | GitHub Repository                             | More Info |
+| ----------------- | --------------------------------------------- | --------- |
+| art               | {{< github "sourcebots/art" >}}               |           |
+| Arturo            | {{< github "sourcebots/Arturo" >}}            |           |
```
Comment (kierdavis) [inline content/codebase/_index.md]: I'd recommend moving this to "utilities". FWIW it's the build system for the servo assembly firmware.

File: content/codebase/_index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+| pcbmedals   | {{< github "sourcebots/pcbmedals" >}}   |           |
+| pkg         | {{< github "sourcebots/pkg" >}}         |           |
+| power-v4-hw | {{< github "sourcebots/power-v4-hw" >}} |           |
+| sym         | {{< github "sourcebots/sym" >}}         |           |
+
+## Misc
+
+| Name              | GitHub Repository                             | More Info |
+| ----------------- | --------------------------------------------- | --------- |
+| art               | {{< github "sourcebots/art" >}}               |           |
```
Comment (kierdavis) [inline content/codebase/_index.md]: Thoughts on moving this into the same category as the websites, and renaming the category to something like "Media"?

Comment (trickeydan) [reply content/codebase/_index.md]: Good idea :thumbsup:

File: content/codebase/_index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+title: Our Codebase
+weight: 1
+---
+
+{{% notice info %}}
+This section is still under development and some parts may be missing.
+{{% /notice %}}
+
+This page contains a summary of the repositories on [Our GitHub][sb-github].
```
Comment (kierdavis) [inline content/codebase/_index.md]: "Our" likely doesn't need to capitalised or included in the hyperlink here.

File: content/codebase/_index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+| ------------- | ----------------------------------------- | --------- |
+| Arturo        | {{< github "sourcebots/Arturo" >}}        |           |
+| barcode-gen   | {{< github "sourcebots/barcode-gen" >}}   |           |
+| kit-testing   | {{< github "sourcebots/kit-testing" >}}   |           |
+| marker-to-pdf | {{< github "sourcebots/marker-to-pdf" >}} |           |
+| pcb-util      | {{< github "sourcebots/pcb-util" >}}      |           |
+| srcomp-kiosk  | {{< github "sourcebots/srcomp-kiosk" >}}  |           |
+| tools         | {{< github "sourcebots/tools" >}}         |           |
+
+[sb-github]: https://github.com/sourcebots
```
Comment (RealOrangeOne) [inline content/codebase/_index.md]: Looks like you can reuse the GitHub shortcode for this link instead

File: content/codebase/_index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+{{% notice info %}}
+This section is still under development and some parts may be missing.
+{{% /notice %}}
+
+This page contains a summary of the repositories on [GitHub][sb-github].
+
+## Administration
+
+| Name                     | GitHub Repository                                    | More Info  |
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply content/codebase/_index.md]: If this table is going to be that generic, I propose we abstract out as much as we can into shortcodes for each row.

Unrelated: I've had quite good success on my own repo calling the GitHub API at build time, which could be fun

**Comment by PeterJCLaw** [comment]
Somewhat serious question: why is this new page better than https://github.com/sourcebots ? It strikes me that it duplicates information to a location where it's likely to get out of date very much faster without providing any additional value.
The additional value it seems to be hoping to provide later appears to be things which should instead be being provided in the READMEs of the individual repos.

Note that I don't think that this is an inherent issue in #18, as I do think there is some high level guidance which would be useful here (see https://github.com/PeterJCLaw/srcomp/wiki for an example of where I'd assume this was headed).

Am I missing something which makes the page as it stands useful?

**Comment by trickeydan** [comment]
Whilst there is duplicate information here, this page is not intended to contain information. It is an index page to link to further pages containing detailed information about how the components of our stack work. For example, it would be bad practice to include detailed information about the internals of robotd in the README for that repository. The volunteer docs exists to explain the more complex internals of our code, rather than just how to use it.
Additionally, there are still repositories in the SourceBots organisation that do not have README documents, or indeed information on how to use them.
In order to decrease the duplication of information, the child pages of this section should link to the README files in order to explain how to use them, whilst the actual pages explain how the modules work. For example, information similar to the presentations that Andy has recently been giving at meetings.
The eventual intention for this page is similar to the wiki that you have linked to, although the detail will be on the child pages. This page is still work in progress, hence the inclusion of the banner which states as such. Merging it has opened up the ability for others to work on the pages below it.

**Comment by PeterJCLaw** [comment]
> For example, it would be bad practice to include detailed information about the internals of robotd in the README for that repository.

While I somewhat agree with this statement, I think you're implying that those details should instead be in the volunteer docs which I definitely disagree with. Documentation for any code should be as close to that code as is reasonably practical; the chances that documentation will get out of date are proportional to their distance from the code.

The level of information which I see as being useful separate from the code repos themselves is that describing how to _combine_ our projects -- this would be akin to information based on their public APIs, which are reasonable to expect are stable over the medium term.

**Comment by trickeydan** [comment]
I will create a separate pull request to create a page for robotd (from my limited knowledge of it, and in an attempt to gain a greater understanding of it). This should help consolidate the scope of this section of the docs.

**Pull request merged**